lua/gatekeeper/staticlib.lua

```diff
@@ -124,6 +124,40 @@ function convert_numa_table_to_array (numa_table)
 	return res
 end
 
+function gk_sol_map (gk_lcores, sol_lcores)
+	local m = {}
+	local sol_allocated = {}
+
+	if #gk_lcores % #sol_lcores ~= 0 then
+		print("Warning: uneven GK-to-SOL blocks assignment");
+	end
+
+	for i, v in ipairs(sol_lcores) do
+		sol_allocated[i] = 0
+	end
+
+	for i1, v1 in ipairs(gk_lcores) do
+		local idx
+		local socket_id = rte_lcore_to_socket_id(v1)
+
+		for i2, v2 in ipairs(sol_lcores) do
+			if rte_lcore_to_socket_id(v2) == socket_id and
+					(idx == nil or sol_allocated[i2] <
+						sol_allocated[idx]) then
+				idx = i2
+			end
+		end
+
+		if idx == nil then
+			error("No SOL block allocated at NUMA node " .. socket_id)
+		end
+
+		m[i1] = idx
+		sol_allocated[idx] = sol_allocated[idx] + 1
+	end
+	return m
```

Comment:
Before returning, iterate over `sol_allocated` and report as warnings any lcore that has zero GK block allocated to it. These warnings will inform operators of lcore waste.